### PR TITLE
TableNG: Column alignment

### LINKED ISSUE #94763: Table: Styling - Column Alignment
Select column alignment.

- Auto
- Left
- Center
- Right

Image

### PR DESCRIPTION
**Pull request by Develer**
**What is this feature?**

This PR provides column alignment feature parity.

**Which issue(s) does this PR fix?**:

https://github.com/user-attachments/assets/a6340bce-bb57-4502-ac21-6e1209cac161

Fixes #94763 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx [hunk 1]
```diff
@@ -49,6 +49,9 @@ export function TableNG(props: TableNGProps) {
     return fieldConfig?.defaults?.custom?.width || 'auto';
   }, [fieldConfig]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  // const columnAlign = fieldConfig?.defaults?.custom?.align;
```
Comment (drew08t) [inline packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx:52]: Remove?

Comment (Develer) [reply packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx]: yes 😄

**Comment by drew08t** [comment]
Works as expected for the table cells. Can we extend this to footer as well?
image

**Comment by Develer** [comment]
> Works as expected for the table cells. Can we extend this to footer as well?

yes, let me do this as follow up

**Comment by drew08t** [comment]
Needs to impact the header alignment as well.

**Comment by adela-almasan** [comment]
Looks good! Same comment as Drew, this also needs to apply to header and footer, and also to other cell types (gauge, sparkline, etc).

https://github.com/user-attachments/assets/7cf668ec-e56d-424b-927e-c42dba9a80d6

File: packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx [hunk 2]
```diff
@@ -191,7 +191,7 @@ export function TableNG(props: TableNGProps) {
         },
         ...(footerOptions?.show && {
           renderSummaryCell() {
-            return <>{getFooterValue(fieldIndex, footerItems)}</>;
+            return <>{getFooterValue(fieldIndex, footerItems, false, justifyColumnContent)}</>;
```
Comment (drew08t) [inline packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx:194]: Shouldn't isCountRowsSet? be dependent on calculation option?

For calculation "Count", when `countRows` is enabled.

**Review by drew08t — APPROVED** [review]
LGTM and works as expected!

**Pull request merged**